src/Client.php

```diff
@@ -829,6 +854,8 @@ protected function calcAuthDigest($user, $password, $type = self::AUTH_BASIC, $d
                         );
                     }
                     $ha2 = md5($this->getMethod() . ':' . $this->getUri()->getPath() . ':' . md5($entityBody));
+                } else {
+                    throw new InvalidArgumentException('Invalid DIGEST auth data');
```

Comment:
There are no tests for that part. It's not used in the code but it's exposed because it's a protected method.

It's a little bit complicated to test it for me, I just added the check because `$ha2` will not be defined.

If we need tests, I think I need help (or time) on it.